controller/state/statedb.go

```diff
@@ -742,14 +742,37 @@ func (s *stateDB) PublishRecordsFrom(ctx context.Context, worker string) error {
 			if err != nil {
 				return err
 			}
-			sig, err := s.PrivKey.Sign([]byte(lnk))
+			c, err := cid.Decode(lnk)
 			if err != nil {
 				return err
 			}
-			c, err := cid.Decode(lnk)
+
+			// check if we should include it.
+			rcrdRdr, err := txContextLoader(ctx, tx)(cidlink.Link{c}, ipld.LinkContext{})
+			if err != nil {
+				return err
+			}
+			tskBuilder := tasks.Type.FinishedTask.NewBuilder()
+			if err := dagjson.Decoder(tskBuilder, rcrdRdr); err != nil {
+				return err
+			}
+			tsk := tskBuilder.Build().(tasks.FinishedTask)
+			if tsk.FieldErrorMessage().Exists() {
+				em := tsk.FieldErrorMessage().Must().String()
+				if strings.Contains(em, "there is an active retrieval deal with") ||
+					strings.Contains(em, "blockstore: block not found") ||
+					strings.Contains(em, "missing permission to invoke") ||
+					strings.Contains(em, "/efs/dealbot") ||
+					strings.Contains(em, "handler: websocket connection closed") {
```

Comment:
use a loop?

Reply:
you mean put these strings in an array and loop through them?

Reply:
Yes :)